script/SingleBatchEditionDemo.s.sol

```diff
@@ -67,18 +46,30 @@ contract SingleBatchEditionDemo is Script, ShowtimeVerifierFixture {
             0xF9984Db6A3bd7044f0d22c9008ddA296C0CC5468  // henryfontanier.eth
         );
 
-        address pointer = SSTORE2.write(recipients);
+        EditionData memory editionData = EditionData({
+            // quotes will need to be escaped:
+            name: unicode'"She gets visions" 👁️',
+            // newlines will need to be escaped
+            description: unicode"Playing in the background:\nKetto by Bonobo 🎶",
+            animationUrl: "",
+            imageUrl: "ipfs://QmSEBhh7A4JKjdRAVEwLGmfF5ckabAUnYVace9KjvyqMZj",
+            royaltyBPS: 10_00,
+            externalUrl: "https://showtime.xyz/nft/polygon/0x1D6378e337f49dA12eEf49Bd1D8de3a1720115f4/0",
+            creatorName: "@AliceOnChain",
+            tags: "art,limited-edition,time-limited"
+        });
 
-        SingleBatchEditionMinter minter = SingleBatchEditionMinter(editionCreator.minter());
+        SignedAttestation memory signedAttestation = signed(pk, creatorAttestation);
 
-        Attestation memory batchAttestation = Attestation({
-            context: address(edition),
-            beneficiary: pointer,
-            validUntil: block.timestamp + 2 minutes,
-            nonce: verifier.nonces(pointer) // expected to be 0
-        });
+        address editionAddress = address(
+            editionFactory.createEdition(
+                editionData,
+                recipients,
+                signedAttestation
+            )
+        );
```

Comment:
> The lack of a minter contract is interesting. I think we may need to remove a non-null constraint on the address we save for it but that's not a huge issue. Probably saves us some gas too no?

In the case of a single batch edition, by the time it's deployed with the factory it's already minted so you're not going to need a minter. 

However maybe now is a good time to think about future ergonomics: we may later introduce a multi-batch edition to go beyond 1k mints, and so these will likely require a minter contract

Reply:
That makes sense.

I brought up the fact that we'd be losing unlimited editions (and have a hard cap on size) with Alex, since this is probably most impactful for gated drops.

Maybe we can have a separate flow for gated drops? Right now regular drops are paid and max out at 1000 for edition size so that's fine. 

For gated drops, we can possibly create + mint an initial set of claimers when one of the following happens:
- If it's an open edition (non-presave/has no web2 deadline), either we reach 1000 claimers or it expires
- If it's a presave drop, when the song is released

In this scenario, we can maybe have two new functions:
- `commitOpenEdition`
  - Store + create edition + mint + register minter contract
- `mintOpenEditionBatch`
  - Store + mint via minter contract

We didn't entirely remove the minter contract piece right? So we can probably still reuse `SingleBatchEdition` and maybe add a new parameter for `freeze_mints`?